tests/test_ui_server.py

```diff
@@ -0,0 +1,65 @@
+import os
+import json
+import unittest
+
+TEST_DATA_PATH = os.path.join(
+    os.path.dirname(__file__), '..', 'data', 'invest-test-data')
+
+
+class EndpointFunctionTests(unittest.TestCase):
+
+    def test_get_vector_colnames(self):
+        from natcap.invest import ui_server
+        test_client = ui_server.app.test_client()
+        # an empty path
+        response = test_client.post('/colnames', json={'vector_path': ''})
+        colnames = json.loads(response.get_data(as_text=True))
+        self.assertEqual(colnames, [])
+        # a vector with one column
+        path = os.path.join(
+            TEST_DATA_PATH, 'aquaculture', 'Input', 'Finfish_Netpens.shp')
+        response = test_client.post('/colnames', json={'vector_path': path})
+        colnames = json.loads(response.get_data(as_text=True))
+        self.assertEqual(colnames, ['FarmID'])
+        # a non-vector file shouldn't raise an error
+        path = os.path.join(TEST_DATA_PATH, 'ndr', 'input', 'dem.tif')
+        response = test_client.post('/colnames', json={'vector_path': path})
+        colnames = json.loads(response.get_data(as_text=True))
+        self.assertEqual(colnames, [])
```

Comment:
It makes sense to me that a non-vector file won't have any column names, but isn't that an error state?  Should that return an error HTTP code as well?

Reply:
I was thinking to make it return the value that's relevant to the UI (an empty list) but you're right that it would make more sense to be an error, and then have the UI handle that error. Probably a `422` status code? (https://stackoverflow.com/questions/7939137/right-http-status-code-to-wrong-input)